src/riak_kv_w1c_worker.erl

```diff
@@ -337,6 +333,22 @@ get_request_record(ReqId, #state{entries=Entries} = _State) ->
         error -> undefined
     end.
 
+%% Utility function for put/2: is this a TS object with its special
+%% requirements or a more traditional KV object?
+%%
+%% When riak_kv_pb_timeseries is driving a put request, it can provide
+%% all of these details directly to async_put/5, but when a tombstone
+%% is being put via riak_kv_delete, these must be extracted from the
+%% object.
+kv_or_ts_details(RObj, {ok, LocalKey}) ->
+    MD  = riak_object:get_metadata(RObj),
+    MD1 = dict:erase(?MD_TS_LOCAL_KEY, MD),
```

Comment:
This is for scenarios where `riak_kv_w1c_worker:put` is invoked by an alternate path (specifically, putting a tombstone during a delete request).